R/utilities-units.R

```diff
@@ -121,7 +121,34 @@ allAvailableDimensions <- function() {
 getDimensionForUnit <- function(unit) {
   validateIsString(unit)
   unit <- encodeUnit(unit)
-  dimensionTask <- getNetTask("DimensionTask")
+  dimensionTask <- getDimensionTask()
   dim <- rClr::clrCall(dimensionTask, "DimensionForUnit", unit)
   ifNotNull(dim, rClr::clrGet(dim, "Name"))
 }
+
+#' Returns a vector containing all units defined in the dimension
+#'
+#' @param dimension Name of dimension for which units should be returned
+#'
+#' @examples
+#' units <- getUnitsForDimension("Mass")
+#' @export
+getUnitsForDimension <- function(dimension) {
+  validateIsString(dimension)
+  dimensionTask <- getDimensionTask()
+  rClr::clrCall(dimensionTask, "AllAvailableUnitNamesForDimension", dimension)
+}
+
+
+#' Return an instance of the .NET Task `DimensionTask`
+#' This is purely for optimization purposes
+#'
+#' @return An instance of the Task
+getDimensionTask <- function() {

```

Comment:
@PavelBal I added your method here. I'd love to know how much of a difference such as small optimization is making

Reply:
To be honest I did not investigate it. Might think of a use case and profile it.

Reply:
Yeah. I am pretty sure this is completely insignificant to be honest. But well why not